### ISSUE TITLE
Unable to delete instance immediately after creation

### ISSUE THREAD

**Issue by markdryan** [01org/ciao#746]
There seems to be a race condition with instance creation.  When writing the BAT tests I create an instance and then try to delete it straight away without first waiting for that instance to become active.  When a I do this I get the following error from ciao-cli:

```


```

If I insert a sleep of 2 seconds between instance creation and deletion I don't get the error.

**Comment by kaccardi** [comment]
@markdryan I don't think I'd call this a bug - it is by design. You can't delete an instance that's "pending". You have to wait till it's either active, or exited or something. Allowing a delete prior to this opens up all kinds of race conditions. I also don't want to queue commands in controller for the same reason. So to me, this is operator error.

**Comment by kaccardi** [comment]
@markdryan I do think status code 500 isn't the right response though.

**Comment by markdryan** [comment]
But isn't this a race condition?  Currently, you need to wait an arbitrary amount of time before you can delete something?

**Comment by kaccardi** [comment]
where would the race condition be? in any automated bat tests? yes - you have to keep checking and timeout after some predefined wait.

**Comment by kaccardi** [comment]
the debug UI had this flow: get status (once time). If user requests a delete on an instance that at that moment in time active or exited, pop up error message and do not send the request. Then the user has to chose when to try again.

**Comment by markdryan** [comment]
RIght.  That's where I'm seeing this.  In the BATs.  If this is by design feel free to close.  I can wait until I get a status in the BATs before deleting.  Although maybe we want to keep it open until we change the error code.
